libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo/api/http/DefaultHttpRequestComposer.kt

```diff
@@ -283,10 +314,11 @@ class DefaultHttpRequestComposer(
         customScalarAdapters: CustomScalarAdapters,
         autoPersistQueries: Boolean,
         sendDocument: Boolean,
+        sendEnhancedClientAwarenessExtensions: Boolean,
```

Comment:
This appears to build without error but with this function being public (I think that's the default visibility when not specified?), is it going to break user code without a default on the param; do users typically call this function directly?

Reply:
This will most likely break the ABI compatibility indeed. How default parameter values work is that they generate a synthetic function that takes an extra int parameter, that extra int being a bitmask of what values are explicitely passed. See https://medium.com/@yi.lu_55329/is-kotlin-default-arguments-a-solution-of-binary-backward-compatibility-374d2c9214d4 for a more in depth explanation.

You can see the binary compatibility validator failed [here](https://github.com/apollographql/apollo-kotlin/actions/runs/15263435025/job/42924993454?pr=6537#step:5:308).

I recommend to make it an extra new method and deprecate the old one. This is a bit painful but it's the only way to preserve backward compatibility.